java/src/jni/h5pDAPLImp.c

```diff
@@ -80,11 +80,10 @@ Java_hdf_hdf5lib_H5_H5Pget_1chunk_1cache(JNIEnv *env, jclass clss, jlong dapl, j
     if (NULL != rdcc_nbytes)
         PIN_LONG_ARRAY(ENVONLY, rdcc_nbytes, nbytesArray, &isCopy,
                        "H5Pget_chunk_cache: nbytesArray array not pinned");
+    if (NULL == nbytesArray)
+        H5_NULL_ARGUMENT_ERROR(ENVONLY, "nbytesArray should not be NULL after pinning");
```

Comment:
The above code is only pinning when the input arguments aren't NULL, so won't this throw an error if NULL is passed in for `rdcc_nbytes`?

Reply:
@byrnHDF  asked me to move it here. I don't normally do much work in the Java wrappers. Which is the correct behavior?

Reply:
In this case it looks like we probably need a block starting at `if (NULL != rdcc_nbytes)` and have the check for `NULL == nbytesArray` inside that block. Someone can call H5.H5Pget_chunk_cache with NULL pointer arguments so that only the non-NULL pointer arguments passed in get populated. Since one could pass a NULL `rdcc_nbytes`, the check for `nbytesArray` being NULL after the PIN call should only happen if we actually tried to pin it.

Reply:
However, there are other issues here in that the below code seems to assume `nbytesArray` is non-NULL and dereferences it.

Reply:
I guess adding this check is better than the current situation that if NULL comes in for `rdcc_nbytes`, the code would crash the JVM. The code here probably still needs to be revised at some point though.

Reply:
So the three jlongArray buffers must exit in order to return information, so the three checks should have the else like:
`    if (NULL != rdcc_w0)
        PIN_DOUBLE_ARRAY(ENVONLY, rdcc_w0, w0Array, &isCopy, "H5Pget_chunk_cache: rdcc_w0 array not pinned");
    else
        H5_NULL_ARGUMENT_ERROR(ENVONLY, "rdcc_w0 should not be NULL ");
    if (NULL != rdcc_nslots)
        PIN_LONG_ARRAY(ENVONLY, rdcc_nslots, rdcc_nslotsArray, &isCopy,
                       "H5Pget_chunk_cache: rdcc_nslots array not pinned");
    else
        H5_NULL_ARGUMENT_ERROR(ENVONLY, "rdcc_nslots should not be NULL ");
    if (NULL != rdcc_nbytes)
        PIN_LONG_ARRAY(ENVONLY, rdcc_nbytes, nbytesArray, &isCopy,
                       "H5Pget_chunk_cache: nbytesArray array not pinned");
    else
        H5_NULL_ARGUMENT_ERROR(ENVONLY, "rdcc_nbytes should not be NULL");
Member
`

Reply:
In the C API, `rdcc_nslots`, `rdcc_nbytes` and `rdcc_w0` are optional. If we don't want them to be optional in Java, then yes we should throw an error if any are NULL.

Reply:
So that is why it doesn't throw error. But why is it checking just one?

Reply:
What should be done then is that the assignments after the call and before done is only done if the array is not NULL;
`        if (rdcc_nslotsArray!=NULL)
              *rdcc_nslotsArray = (jlong)rdcc_nslots_t;
        if (nbytesArray!=NULL)
              *nbytesArray      = (jlong)nbytes_t;
`

Reply:
Yes I think the code needs some revising depending on what we want the semantics of the Java API to be